ocaml/xapi/xapi_pif.ml

```diff
@@ -495,17 +496,27 @@ let scan ~__context ~host =
 			(List.map fst t.device_to_mac_table)
 			(List.map snd t.pif_to_device_table) in
 
-	let non_managed_devices =
+	let non_managed_devices, disallow_unplug_devices =
 		if Sys.file_exists !Xapi_globs.non_managed_pifs then
 			try
 				let output, _ = Forkhelpers.execute_command_get_output !Xapi_globs.non_managed_pifs [] in
-				String.split_f String.isspace output
+				let dsplit = String.split '\n' output in
+				match dsplit with
```

Comment:
Anoob has already pointed out himself that this match can probably be simplified, and is looking into it.